### PR TITLE
Replace `cached_property` with `backports.cached_property`

### PR DESCRIPTION
**Pull request by RishiKumarRay**
updated dependency to backports.cached_property

## Description
Removed older cached_property
installed new backports.cached_property
updated references

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

Closes #1517 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Rishi Kumar Ray for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/field.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
     overload,
 )
 
-from cached_property import cached_property  # type: ignore
+from backports.cached_property import backports.cached_property  # type: ignore
```
Comment (BryceBeagle) [inline strawberry/field.py:21]: This is not the proper way to import the `cached_property` symbol. You need to do:
```suggestion
from backports.cached_property import cached_property
```

Did you test these changes?

**Review by BryceBeagle — DISMISSED** [review]

**Comment by RishiKumarRay** [comment]
@BryceBeagle can you please review the PR now

File: strawberry/types/fields/resolver.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from inspect import isasyncgenfunction, iscoroutinefunction
 from typing import Callable, Dict, Generic, List, Mapping, Optional, TypeVar, Union
 
-from cached_property import cached_property  # type: ignore
+from backports.cached_property import cached_property  # type: ignore
```
Comment (patrick91) [inline strawberry/types/fields/resolver.py:9]:
```suggestion
from backports.cached_property import cached_property 
```

**Comment by RishiKumarRay** [comment]
@patrick91 I have updated your changes. Please do the review

**Comment by patrick91** [comment]
@RishiKumarRay thanks! We are missing the `RELEASE.md` file as suggested by the bot above, can you add it?

There's also a couple of failures related to mypy, can you add a type ignore on `strawberry/field.py:135`?

**Comment by RishiKumarRay** [comment]
@patrick91 Hi, Patrick I am sorry to say but i have messed up with my repository that i forked and I also don't know how to add this , I was only able to commit your suggestions , I feel i won't be able to do

**Comment by patrick91** [comment]
> @patrick91 Hi, Patrick I am sorry to say but i have messed up with my repository that i forked and I also don't know how to add this , I was only able to commit your suggestions , I feel i won't be able to do

if you want I can try to help with that on discord :)

**Comment by RishiKumarRay** [comment]
Yeah please let me message you there @patrick91

**Comment by patrick91** [comment]
> Yeah please let me message you there @patrick91

I think you can message me here -> `patrick.py#0235`

or you can join the strawberry discord http://strawberry.rocks/discord

**Review by patrick91 — APPROVED** [review]
Nice one! Thanks 😊

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**